test/fuzz/fuzz_doc_symbols.cc

```diff
@@ -0,0 +1,77 @@
+#include "core/Unfreeze.h"
+#include "main/lsp/wrapper.h"
+#include "main/pipeline/pipeline.h"
+#include "payload/payload.h"
+#include "spdlog/sinks/stdout_sinks.h"
+#include "test/helpers/MockFileSystem.h"
+#include "test/helpers/lsp.h"
+#include "test/helpers/position_assertions.h"
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+
+const auto console = spdlog::stdout_logger_mt("console");
+const auto typeErrors = spdlog::stdout_logger_mt("typeErrors");
+const auto rootPath = "/tmp";
+const auto rootUri = fmt::format("file://{}", rootPath);
+
+sorbet::realmain::options::Options mkOpts() {
+    sorbet::realmain::options::Options opts;
+    opts.fs = std::make_shared<sorbet::test::MockFileSystem>(rootPath);
+    opts.noStdlib = true;
+    opts.lspDocumentSymbolEnabled = true;
+    opts.rawInputDirNames.emplace_back(rootPath);
+    return opts;
+}
+
+std::unique_ptr<sorbet::core::GlobalState> mkGlobalState(const sorbet::realmain::options::Options &opts,
+                                                         std::unique_ptr<sorbet::KeyValueStore> &kvstore) {
+    auto gs = std::make_unique<sorbet::core::GlobalState>(
+        (std::make_shared<sorbet::core::ErrorQueue>(*typeErrors, *console)));
+    sorbet::payload::createInitialGlobalState(gs, opts, kvstore);
+    gs->errorQueue->ignoreFlushes = true;
+    return gs;
+}
+
+sorbet::realmain::lsp::LSPWrapper mkLSPWrapper() {
+    std::unique_ptr<sorbet::KeyValueStore> kvstore;
+    auto opts = mkOpts();
+    static const auto commonGs = mkGlobalState(opts, kvstore);
+
+    // TODO why?
```

Comment:
The other fuzzer uses this because it's cheaper to `deepCopy` a GlobalState than it is to initialize one. I don't know why it's in its own scope though.

Reply:
I guess I was wondering why we make a `commonGs` and then copy it, never using `commonGs` again. Could we just do `const auto gs = mkGlobalState(opts, kvstore)`?

Reply:
`commonGs` is `static`, so it'll persist across fuzzer runs on different input in the same process. It's the only shared state across runs, but it should be safe since we don't mutate it -- we copy it every time.

Reply:
In other words, in one process, `mkGlobalState` will only run once but will run thousands of fuzzer inputs.

Reply:
(If `mkGlobalState` is really fast/the same cost as a `deepCopy`, then we can do as you said.)

Reply:
Gotcha. I'll be honest, I'm hazy on the semantics of `static` (I feel like it does 10 different things?). Thanks!